### PR TITLE
feat: Now taking argument as variables (making them mutable)

### PR DESCRIPTION
**Pull request by EdenComp**

### CODE REVIEW

File: src/Dreamberd/Compilation/Compile.hs [hunk 1]
```diff
@@ -55,7 +54,7 @@ compileReturn _ Nothing = Right [VM.Push VM.Void, VM.Ret]
 compileFunction :: [String] -> String -> [String] -> [AST.AstNode] -> Either String [VM.Insts]
 compileFunction params name args body =
     compileNodes (args ++ params) body []
-        >>= \bodyInsts -> Right [VM.DefineEnv name False $ Just $ VM.Function bodyInsts]
+        >>= \bodyInsts -> Right [VM.DefineEnv name False (Just $ VM.Function (concatMap (\nb -> [VM.PushArg nb, VM.DefineEnv (args !! nb) False Nothing]) [0..(length args - 1)] ++ bodyInsts ++ map VM.EraseEnv args))]
```
Comment (Croos3r) [inline src/Dreamberd/Compilation/Compile.hs:57]: damn

**Review by Croos3r — APPROVED** [review]
me when you show up

File: src/Dreamberd/Compilation/Compile.hs (hunk 1 above)
Comment (EdenComp) [reply src/Dreamberd/Compilation/Compile.hs]: éwé

**Review by Tomi-Tom — APPROVED** [review]
Sheesh

**Pull request merged**